sycl/include/CL/sycl/item.hpp

```diff
@@ -21,6 +21,10 @@ __SYCL_INLINE_NAMESPACE(cl) {
 namespace sycl {
 namespace detail {
 class Builder;
+template <typename TransformedArgType, int Dims, typename KernelType>
```

Comment:
From the changes it is not clear why RoundedRangeKernel needs to be friend to id/item/range classes?
If they really are, would you please 1-2 line(s) comment before these friendship declarations?

Reply:
It's because new classes `RoundedRangeKernel` and `RoundedRangeKernelWithKH` access `set_allowed_range` methods of `id/item/range` that are private. Previously those accesses were made in the `handler` class. Will update the description.

Reply:
Thank you for the info. The classes 'id' and 'item' have that 'set_allowed_range' method.
Would you add a 1 line comment here. Something like:
// Friend to get access to private method set_allowed_range().

2) The class 'range' does not have method 'set_allowed_range'. Does it also need to have those 2 new functor/classes as a friend? (see range.hpp changed below).